extensions/HyperVExtension/src/HyperVExtension/Services/VMGalleryService.cs

```diff
@@ -79,9 +79,10 @@ public async Task<VMGalleryImageList> GetGalleryImagesAsync()
                     var totalSizeOfDisk = await _downloaderService.GetHeaderContentLength(new Uri(image.Disk.Uri), cancellationTokenSource.Token);
                     if (ulong.TryParse(image.Requirements.DiskSpace, CultureInfo.InvariantCulture, out var requiredDiskSpace))
                     {
-                        image.Disk.SizeInBytes = (ulong)totalSizeOfDisk;
+                        image.Disk.ArchiveSizeInBytes = (ulong)totalSizeOfDisk;
+                        image.Disk.ExtractedFileRequiredFreeSpace = requiredDiskSpace + image.Disk.ArchiveSizeInBytes;
```

Comment:
See my reply to Darren in the `VMImageSelectionValidator.cs` about the extracted file required size.  

> requiredDiskSpace + image.Disk.ArchiveSizeInBytes 

does not equal the size of the extracted file in practice. There are two things I can do here as it would be preferred to have this information without extracting the file first.

1. Update the Vm gallery pipeline to explicitly include the size of the archive file or files.
2. Or have a hard coded list with these that we'd have to maintain ourselves.

I prefer the first one. I'll follow up with Kevins team and update the PR.

Reply:
Spoke to Anis, and created this issue: #3714 . For now we'll use hardcoded values for the size of the extracted vhdx files until we can update the gallery directly. I updated the PR accordingly